### PR TITLE
Improve Empty audience error message

### LINKED ISSUE #1410: Improve error message when 'aud' claim is empty
I'm currently attempting to verify a JWT received from Cognito, which [does not include the aud claim](https://stackoverflow.com/q/53148711/10100823).  As [the aud claim is optional](https://tools.ietf.org/html/rfc7519#section-4.1.3), this should be fine.  However, I'm receiving the error:

```
Error validating identity token: Microsoft.IdentityModel.Tokens.SecurityTokenInvalidAudienceException: IDX10214: Audience validation failed. Audiences: 'empty'. Did not match: validationParameters.ValidAudience: '[CENSORED]' or validationParameters.ValidAudiences: 'null'.
   at Microsoft.IdentityModel.Tokens.Validators.ValidateAudience(IEnumerable`1 audiences, SecurityToken securityToken, TokenValidationParameters validationParameters)
   at System.IdentityModel.Tokens.Jwt.JwtSecurityTokenHandler.ValidateTokenPayload(JwtSecurityToken jwtToken, TokenValidationParameters validationParameters)
   at System.IdentityModel.Tokens.Jwt.JwtSecurityTokenHandler.ValidateToken(String token, TokenValidationParameters validationParameters, SecurityToken& validatedToken)
```

It looks like [that error is being thrown here](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/f5628e689e97b80fba3efe81fc33903076c97ad5/src/Microsoft.IdentityModel.Tokens/Validators.cs#L126), and 'empty' is just what`SerializeAsSingleCommaDelimitedString` gives back for [an empty list](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/f5628e689e97b80fba3efe81fc33903076c97ad5/src/Microsoft.IdentityModel.Tokens/Utility.cs#L45).  Per discussion on #1402 , I'm disabling `validationParameters.ValidateAudience` as a workaround.

As the default PII settings won't show the 'empty' `aud` by default, the error message makes it seem as though something is being set as a default value, and adds some digging to figure out what's going on.  A clearer error message when `aud` is required but not present would be appreciated.

An example could just be "missing required claim: aud"

### PR DESCRIPTION
**Pull request by keegan-caruso**
Fixes #1410

### CODE REVIEW

File: src/Microsoft.IdentityModel.Tokens/Validators.cs [hunk 1]
```diff
@@ -129,40 +146,58 @@ public static void ValidateAudience(IEnumerable<string> audiences, SecurityToken
...
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private static bool AudiencesMatchIgnoringTrailingSlash(string tokenAudience, string validAudience)
+        {
+            int length;
```
Comment (RojaEnnam) [inline src/Microsoft.IdentityModel.Tokens/Validators.cs]: length can be assigned -1 here itself ?

File: src/Microsoft.IdentityModel.Tokens/Validators.cs [hunk 2]
```diff
@@ -129,40 +146,58 @@ public static void ValidateAudience(IEnumerable<string> audiences, SecurityToken
...
-                    { InvalidAudience = Utility.SerializeAsSingleCommaDelimitedString(audiences) });
+            return false;
+        }
+
+        private static bool AudiencesMatch(TokenValidationParameters validationParameters, string tokenAudience, string validAudience)
+        {
+            if (validAudience.Length == tokenAudience.Length)
+            {
+                if (string.Equals(validAudience, tokenAudience, StringComparison.Ordinal))
+                {
```
Comment (RojaEnnam) [inline src/Microsoft.IdentityModel.Tokens/Validators.cs]: nit: remove braces.

File: src/Microsoft.IdentityModel.Tokens/Validators.cs (hunk 1 above)
Comment (keegan-caruso) [reply src/Microsoft.IdentityModel.Tokens/Validators.cs]: Good call, that is easier to read. Originally just taking a naïve conversion from the nested ternary

**Review by RojaEnnam — APPROVED** [review]

**Review by GeoK — APPROVED** [review]

**Pull request merged**